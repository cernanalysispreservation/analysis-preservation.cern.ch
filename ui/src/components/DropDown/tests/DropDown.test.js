import React from "react";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import DropDown from "../DropDown";
import AddIcon from "grommet/components/icons/base/FormAdd";
import MinusIcon from "grommet/components/icons/base/FormSubtract";

configure({ adapter: new Adapter() });

let wrapper;
const items = [{ service: "github", status: 200 }];
beforeEach(() => {
  wrapper = mount(<DropDown items={items} />);
});
afterEach(() => {
  wrapper.unmount();
});

describe("DropDown Suite", () => {
  it("Take a snapshot", () => {
    const shallowDropDown = shallow(<DropDown />);
    expect(shallowDropDown).toMatchSnapshot();
  });

  it("Minimize the DropDown", () => {
    expect(wrapper.find(MinusIcon)).toHaveLength(1);
    wrapper.setState({ open: false });
    expect(wrapper.find(MinusIcon)).toHaveLength(0);
  });

  it("Expand the DropDown", () => {
    expect(wrapper.find(AddIcon)).toHaveLength(0);
    wrapper.setState({ open: false });
    expect(wrapper.find(AddIcon)).toHaveLength(1);
  });
});
