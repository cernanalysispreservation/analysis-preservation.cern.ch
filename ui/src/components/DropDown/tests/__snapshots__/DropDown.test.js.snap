// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown Suite Take a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropDown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "announce": false,
      "children": Array [
        <Box
          align="center"
          announce={false}
          direction="row"
          flex={true}
          focusable={true}
          onClick={[Function]}
          pad="none"
          responsive={true}
          size="large"
          style={
                    Object {
                              "border": "1px solid black",
                            }
          }
          tag="div"
>
          <FormSubtract
                    responsive={true}
          />
          <Paragraph
                    align="center"
          />
</Box>,
        <DropDownList
          items={undefined}
          open={true}
/>,
      ],
      "direction": "column",
      "focusable": true,
      "margin": Object {
        "vertical": "small",
      },
      "pad": "none",
      "responsive": true,
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "center",
          "announce": false,
          "children": Array [
            <FormSubtract
              responsive={true}
/>,
            <Paragraph
              align="center"
/>,
          ],
          "direction": "row",
          "flex": true,
          "focusable": true,
          "onClick": [Function],
          "pad": "none",
          "responsive": true,
          "size": "large",
          "style": Object {
            "border": "1px solid black",
          },
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "responsive": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": "center",
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "items": undefined,
          "open": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "announce": false,
        "children": Array [
          <Box
            align="center"
            announce={false}
            direction="row"
            flex={true}
            focusable={true}
            onClick={[Function]}
            pad="none"
            responsive={true}
            size="large"
            style={
                        Object {
                                    "border": "1px solid black",
                                  }
            }
            tag="div"
>
            <FormSubtract
                        responsive={true}
            />
            <Paragraph
                        align="center"
            />
</Box>,
          <DropDownList
            items={undefined}
            open={true}
/>,
        ],
        "direction": "column",
        "focusable": true,
        "margin": Object {
          "vertical": "small",
        },
        "pad": "none",
        "responsive": true,
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "center",
            "announce": false,
            "children": Array [
              <FormSubtract
                responsive={true}
/>,
              <Paragraph
                align="center"
/>,
            ],
            "direction": "row",
            "flex": true,
            "focusable": true,
            "onClick": [Function],
            "pad": "none",
            "responsive": true,
            "size": "large",
            "style": Object {
              "border": "1px solid black",
            },
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "responsive": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "align": "center",
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "items": undefined,
            "open": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
